### PR TITLE
WoWGuid: Add functions and minor cleanup

### PR DESCRIPTION
**Pull request by Zallios**
- add getGuidHighPart
- add getHigh
- add is-functions for object types

### CODE REVIEW

File: src/shared/WoWGuid.h [hunk 1]
```diff
@@ -154,25 +198,52 @@ class SERVER_DECL WoWGuid
...
-        const uint32_t lowGuid = *(reinterpret_cast<const uint32_t*>(&oldguid));
-        return lowGuid & 0x00FFFFFF;
-    }
-    uint32_t getGuidHigh() const { return static_cast<uint32_t>(oldguid >> 32); }
-        uint64 GetOldGuid() const { return oldguid; }
+        uint32_t getGuidLow() const { return static_cast<uint32_t>(m_rawGuid); }
+        uint32_t getGuidLowPart() const
+        {
+            const uint32_t lowGuid = *(reinterpret_cast<const uint32_t*>(&m_rawGuid));
+            return lowGuid & 0x00FFFFFF;
```
Comment (Zyres) [inline src/shared/WoWGuid.h]: Magic number replace with enum value;

File: src/shared/WoWGuid.h [hunk 2]
```diff
@@ -154,25 +198,52 @@ class SERVER_DECL WoWGuid
...
+        }
+
+        uint32_t getGuidHigh() const { return static_cast<uint32_t>(m_rawGuid >> 32); }
+        int32_t getGuidHighPart() const
+        {
+            const uint32_t highGuid = *(reinterpret_cast<const uint32_t*>(&m_rawGuid) + 1);
+            return highGuid & 0xFFF00000;
```
Comment (Zyres) [inline src/shared/WoWGuid.h]: Magic number replace with enum value;

File: src/shared/WoWGuid.h [hunk 3]
```diff
@@ -204,7 +275,7 @@ class SERVER_DECL WoWGuid
             for(uint32 x = 0; x < 8; x++)
             {
 
-                uint8 p = ((uint8*)&oldguid)[x];
+                uint8 p = reinterpret_cast<uint8*>(&m_rawGuid)[x];
```
Comment (Zyres) [inline src/shared/WoWGuid.h]: uint8_t

File: src/shared/WoWGuid.h [hunk 4]
```diff
@@ -222,42 +293,42 @@ class SERVER_DECL WoWGuid
 
             if(guidmask & 0x01)  //1
             {
-                oldguid |= ((uint64)guidfields[j]);
+                m_rawGuid |= static_cast<uint64>(guidfields[j]);
```
Comment (Zyres) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h [hunk 5]
```diff
@@ -84,21 +131,18 @@ class SERVER_DECL WoWGuid
             Clear();
         }
 
-        WoWGuid(uint64 guid)
+        WoWGuid(uint64_t guid)
         {
-            Clear();
-            Init((uint64)guid);
+            Init(static_cast<uint64>(guid));
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h [hunk 6]
```diff
@@ -154,25 +198,52 @@ class SERVER_DECL WoWGuid
...
-        bool operator !=(uint64 someval) const { return (oldguid != someval); }
-        uint64 operator &(uint64 someval) const { return (oldguid & someval); }
-        uint64 operator &(unsigned int someval) const { return (oldguid & someval); }
-        operator bool() { return (oldguid > 0); }
-        operator uint64() { return oldguid; }
-        void operator =(uint64 someval) { Clear(); Init((uint64)someval); }
+        bool operator !() const { return (!m_rawGuid); }
+        bool operator ==(uint64 someval) const { return (m_rawGuid == someval); }
+        bool operator !=(uint64 someval) const { return (m_rawGuid != someval); }
+        uint64 operator &(uint64 someval) const { return (m_rawGuid & someval); }
```
Comment (Appled) [inline src/shared/WoWGuid.h]: shouldn't these three lines here be uint64_t?

File: src/shared/WoWGuid.h [hunk 7]
```diff
@@ -154,25 +198,52 @@ class SERVER_DECL WoWGuid
...
+        uint64 operator &(unsigned int someval) const { return (m_rawGuid & someval); }
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h [hunk 8]
```diff
@@ -154,25 +198,52 @@ class SERVER_DECL WoWGuid
...
+        operator bool() { return (m_rawGuid > 0); }
+        operator uint64() { return m_rawGuid; }
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h [hunk 9]
```diff
@@ -154,25 +198,52 @@ class SERVER_DECL WoWGuid
...
+        void operator =(uint64 someval) { Clear(); Init(static_cast<uint64>(someval)); }
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h [hunk 10]
```diff
@@ -222,42 +293,42 @@ class SERVER_DECL WoWGuid
...
             if(guidmask & 0x01)  //1
             {
-                oldguid |= ((uint64)guidfields[j]);
+                m_rawGuid |= static_cast<uint64>(guidfields[j]);
                 j++;
             }
             if(guidmask & 0x02)  //2
             {
-                oldguid |= ((uint64)guidfields[j] << 8);
+                m_rawGuid |= (static_cast<uint64>(guidfields[j]) << 8);
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h (hunk 1 above)
Comment (schnek) [reply src/shared/WoWGuid.h]: LOWGUID_ENTRY_MASK = 0x00FFFFFF

File: src/shared/WoWGuid.h [hunk 11]
```diff
@@ -222,42 +293,42 @@ class SERVER_DECL WoWGuid
...
                 j++;
             }
             if(guidmask & 0x04) //4
             {
-                oldguid |= ((uint64)guidfields[j] << 16);
+                m_rawGuid |= (static_cast<uint64>(guidfields[j]) << 16);
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h [hunk 12]
```diff
@@ -222,42 +293,42 @@ class SERVER_DECL WoWGuid
...
                 j++;
             }
             if(guidmask & 0x08)  //8
             {
-                oldguid |= ((uint64)guidfields[j] << 24);
+                m_rawGuid |= (static_cast<uint64>(guidfields[j]) << 24);
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h [hunk 13]
```diff
@@ -222,42 +293,42 @@ class SERVER_DECL WoWGuid
...
                 j++;
             }
             if(guidmask & 0x10) //16
             {
-                oldguid |= ((uint64)guidfields[j] << 32);
+                m_rawGuid |= (static_cast<uint64>(guidfields[j]) << 32);
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h [hunk 14]
```diff
@@ -222,42 +293,42 @@ class SERVER_DECL WoWGuid
...
                 j++;
             }
             if(guidmask & 0x20) //32
             {
-                oldguid |= ((uint64)guidfields[j] << 40);
+                m_rawGuid |= (static_cast<uint64>(guidfields[j]) << 40);
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h (hunk 2 above)
Comment (schnek) [reply src/shared/WoWGuid.h]: HIGHGUID_TYPE_MASK				= 0xFFF00000

File: src/shared/WoWGuid.h [hunk 15]
```diff
@@ -222,42 +293,42 @@ class SERVER_DECL WoWGuid
...
                 j++;
             }
             if(guidmask & 0x40) //64
             {
-                oldguid |= ((uint64)guidfields[j] << 48);
+                m_rawGuid |= (static_cast<uint64>(guidfields[j]) << 48);
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

File: src/shared/WoWGuid.h [hunk 16]
```diff
@@ -222,42 +293,42 @@ class SERVER_DECL WoWGuid
...
                 j++;
             }
             if(guidmask & 0x80)  //128
             {
-                oldguid |= ((uint64)guidfields[j] << 56);
+                m_rawGuid |= (static_cast<uint64>(guidfields[j]) << 56);
```
Comment (Appled) [inline src/shared/WoWGuid.h]: uint64_t

**Pull request merged**